### PR TITLE
Included functionality for burst, continuous and duo/virtuoso

### PR DESCRIPTION
**Pull request by sccrosby**
Included functionality for burst, continuous and duo/virtuoso.

Temperature from duo get's parsed same as pressure bursts.

Three lines have been added to instrument config files 
instrument_type: 'rbr_duo'     #(rbr_virtuoso,rbr_duo) 
recording_type: 'continuous'   # (burst, continuous)
wave_interval: 1800              # [sec] interval for wave statistics for continuous data

If lines are missing default is Virtuoso in Burst mode.

Tested for
- RBRduo continuous. 

Untested for
- virtuoso burst (recommend this be checked, can you check Dan?)
- virtuoso continuous
- rbrduo burst

### CODE REVIEW

**Comment by dnowacki-usgs** [comment]
Sean, this looks like a great addition. I would like to get this merged into stglib. 

A few thoughts: 

* T_28 should go into b-cal and not s-a. B is for burst, S is for statistics. So burst timeseries should go in the burst file. You should drop T_28 before writing to the s-a file. You can see where I do this for the other vars in `nc_waves.py`.
* I like that different functions are called for the different instrument types but it looks like there is a lot of repeated code in each. Could some of the redundant code be in its own utility function?

You can create additional commits to this branch and they will automatically update the PR.

File: stglib/core/utils.py [hunk 1]
```diff
@@ -744,6 +745,79 @@ def loadmat(filename):
...
+    '''
+    try:
+        samplingperiod = conn.execute(
+            "select samplingperiod from schedules").fetchall()[0][0]
+    except sqlite3.OperationalError:
+        samplingperiod = conn.execute(
+            "select samplingperiod from continuous").fetchall()[0][0]
+
+    samplingrate = round(1000/samplingperiod)   #convert from ms to sec
+    samplingperiod = 1./samplingrate     #period from rate (Hz)
```
Comment (dnowacki-usgs) [inline stglib/core/utils.py]: Why are you redefining `samplingperiod` here? I would suggest you get `samplingperiod` from the SQL query, then adjust its units to seconds, and then define the rate by the inverse of that value. Here you are getting at it in a roundabout way.

File: stglib/rsk/rsk2cdf.py [hunk 2]
```diff
@@ -41,49 +41,37 @@ def rsk_to_xr(metadata):
...
-    ds.attrs['serial_number'] = str(conn.execute(
-        "select serialID from instruments").fetchall()[0][0])
-    ds.attrs['INST_TYPE'] = 'RBR Virtuoso d|wave'
-
-    conn.close()
-
+           'this may take a while for large datasets') % rskfile)    
+
+     #Assume RBRvirtuoso in burst mode if no attrs
+    if 'instrument_type' not in ds.attrs.keys():
```
Comment (dnowacki-usgs) [inline stglib/rsk/rsk2cdf.py]: I prefer this logic without `.keys()`: e.g., `if 'instrument_type' not in ds.attrs`, here and elsewhere

File: stglib/core/utils.py [hunk 3]
```diff
@@ -744,6 +745,79 @@ def loadmat(filename):
     data = spio.loadmat(filename, struct_as_record=False, squeeze_me=True)
     return _check_keys(data)
 
+def read_samplingrates_continuous(ds,conn):
```
Comment (dnowacki-usgs) [inline stglib/core/utils.py]: Can you move these two new functions up above `def read_globalatts`? I like to keep the matlab and a few other general purpose functions at the end.

File: stglib/core/utils.py (hunk 1 above)
Comment (sccrosby) [reply stglib/core/utils.py]: Simplified, thanks

File: stglib/rsk/rsk2cdf.py (hunk 2 above)
Comment (sccrosby) [reply stglib/rsk/rsk2cdf.py]: changes

File: stglib/core/utils.py (hunk 3 above)
Comment (sccrosby) [reply stglib/core/utils.py]: done

**Comment by sccrosby** [comment]
Nice! Thanks for your guidance.

**Comment by dnowacki-usgs** [comment]
PS I found a few errors when testing on a non-continuous file. So make sure you pull the latest version from GH when running your code, thanks!

**Comment by sccrosby** [comment]
Oh good to know! Sorry about those, thanks for fixing!

**Pull request merged**